system-addon/content-src/components/StartupOverlay/StartupOverlay.jsx

```diff
@@ -0,0 +1,87 @@
+import {actionCreators as ac} from "common/Actions.jsm";
+import {connect} from "react-redux";
+import {injectIntl} from "react-intl";
+import React from "react";
+
+export class _StartupOverlay extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.locations = {};
+    this.onInputChange = this.onInputChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+    this.clickSkip = this.clickSkip.bind(this);
+    this.initScene = this.initScene.bind(this);
+
+    this.state = {emailInput: ""};
+  }
+
+  initScene(element) {
+    document.body.dataset.welcome = "true";
+    this.scene = element;
+    // Timout to allow the scene to render once before attaching the attribute
+    // to trigger the animation.
+    setTimeout(() => {
+      this.scene.dataset.content = "true";
```

Comment:
Instead of using data attributes here, can you not just set state here and then add a class in the render function based on the state? And then in the css just check for the class?
```js
this.setState({content: true});
```
and then do:
```html
<div className={"firstrun-scene ${this.state.content ? "show" : "" } "} ref={this.initScene}>
```
Same with `sign-in` data attribute. I say this because generally data attributes are used for stuff that needs actual data, whereas these seem to just be either true or false, which is just as good as adding a class or not.

We do this pattern a bunch: https://github.com/mozilla/activity-stream/blob/master/system-addon/content-src/components/Card/Card.jsx#L142

Reply:
Yeah might as well just make it a div